### DISCUSSION TITLE
How to send a wav file to the browser?

### DISCUSSION THREAD

**Discussion by shyam3089** [miguelgrinberg/Flask-SocketIO#1976] [Q&A] [answered]
I'm working on a browser based voice assistant and trying to play a wav audio in the browser emited by the soicketIO as a response from the server. Here I'm receiving the data as multiple buffers but it takes a long time. Also that data is not playable by JS. What's the correct way to send and receive an audio file using socketIO? 
I don't want to use the Flask's native way to send file.

Server:
```
with open( path + file, "rb") as fwav:
    logging.info("Sending data")
    data = fwav.read(1024)
    while data:
        data = fwav.read(1024)
        socketio.emit('playAudio', {'data': data})
fwav.close()
```

Client:

```
socket.on('playAudio', function (data) {
    console.log("getting audio data", data);
    audioObj = new Audio(data);
    audioObj.play();
});
```

**Comment by miguelgrinberg** [comment, marked as answer]
This isn't a question that is related to Socket.IO. What you need to do is extract the raw audio from the wav file, which is what the Audio object in the browser uses. But I suggest you find a dedicated resource to learn all the details, since neither the server nor the client side are as simplistic as you have them in the code above.
